logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/session/SessionScope.kt

```diff
@@ -1,11 +1,11 @@
 package com.wire.kalium.logic.feature.session
 
 import com.wire.kalium.logic.data.session.SessionRepository
-import com.wire.kalium.logic.feature.conversation.GetConversationsUseCase
+import com.wire.kalium.logic.feature.auth.GetSessionsUseCase
 
 class SessionScope(
-    sessionRepository: SessionRepository
+    private val sessionRepository: SessionRepository
 ) {
-    val allSessions = GetAllSessionsUseCase(sessionRepository)
-    val saveSession = SaveSessionUseCase(sessionRepository)
+    val allSessions get() = GetSessionsUseCase(sessionRepository)
+    val saveSession get() = SaveSessionUseCase(sessionRepository)
```

Comment:
What would be the main gain of creating a new use case instance every time?

Reply:
@gongracr There's no need to keep a reference to one. 

If we stop using them, they can release whatever they're using internally (down to caches, persistence/storage or other IO) and free resources.